### PR TITLE
Upgrade webpacker to v4.

### PR DESCRIPTION
**Pull request by connorshea**
~~I'm not able to get it working with the CSS, but the JavaScript does seem to work fine.~~ Works now!

### CODE REVIEW

File: config/webpacker.yml [hunk 1]
```diff
@@ -65,5 +89,8 @@ production:
   # Production depends on precompilation of packs prior to booting for performance.
   compile: false
 
+  # Extract and emit a css file
+  extract_css: true
```
Comment (maschwenk) [inline config/webpacker.yml]: Probably don't need this given you've added as a default

File: config/webpack/environment.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
 const vue = require('./loaders/vue')
-environment.loaders.append('vue', vue)
 
-// resolve-url-loader must be used before sass-loader
+environment.plugins.prepend('VueLoaderPlugin', new VueLoaderPlugin())
+environment.loaders.prepend('vue', vue)
+
 environment.loaders.get('sass').use.splice(-1, 0, {
   loader: 'resolve-url-loader'
 });
```
Comment (maschwenk) [inline config/webpack/environment.js]: You could probably replace these with prepend or append

Comment (connorshea) [reply config/webpack/environment.js]: I probably should, but I don't actually use the resolve-url-loader currently, so I can probably get away with removing this entirely instead. I tried replacing this with prepend but it didn't quite work as expected.

For now I'll leave it in.

File: config/webpacker.yml (hunk 1 above)
Comment (connorshea) [reply config/webpacker.yml]: Good catch, thanks :)

File: postcss.config.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  plugins: [
```
Comment (sshyam-gupta) [inline postcss.config.js:2]: I don't find these new packages in package.json. Is it built-in provided with React on Rails?

**Pull request merged**